src/libostree/ostree-repo-commit.c

```diff
@@ -1859,6 +2022,9 @@ ostree_repo_write_content (OstreeRepo       *self,
 {
   /* First, if we have an expected checksum, see if we already have this
    * object.  This mirrors the same logic in ostree_repo_write_metadata().
+   *
+   * If we’re writing out size metadata, fall through to the write_content_object()
+   * call anyway, since we need all the objects to end up in self->object_sizes.
```

Comment:
Yeah, looks good. BTW, if you do upstream this, it would be nice to try to make the condition simpler. Like, only set `generate_sizes` if the mode is archive-z2. Then you can just check that and don't also need the separate `indexable` variable.

Do you know why the sizes are only generated on archive-z2? I didn't work on this feature and never noticed it until this review.